### PR TITLE
SES-212 - Always show delivery status of last sent message - FINAL!

### PR DESCRIPTION
**Pull request by alansley**
### Contributor checklist
- [X] I have tested my contribution on these devices:
 * Samsung S9+ Android 9 / API 28
 * Virtual Pixel 3a Android 9 / API 28
 * Virtual Pixel 3a Android 14 / API 34
- [X] My contribution is fully baked and ready to be merged as is
- [X] I ensure that all the open issues my contribution fixes are mentioned in the commit message of my first commit using the `Fixes #1234` [syntax](https://help.github.com/articles/closing-issues-via-commit-messages/)

----------

### Description
Fixes the issue that the delivery status of the last sent message by the specific user logged in should always be visible.

GitHub issue: https://github.com/oxen-io/session-android/issues/1408
Jira ticket: https://optf.atlassian.net/browse/SES-212

Tested by sending messages between virtual and physical devices across multiple versions of Android, and sending/receiving messages to/from the desktop client. This time it works with images and things in the MmsDb (which have separate indices to text messages in the SmsDb)!

### CODE REVIEW

File: app/src/main/java/org/thoughtcrime/securesms/conversation/v2/messages/VisibleMessageView.kt [hunk 1]
```diff
@@ -219,25 +220,20 @@ class VisibleMessageView : LinearLayout {
...
-            )
-            binding.messageStatusImageView.isVisible = (
-                iconID != null && (
-                    !message.isSent ||
-                    message.id == lastMessageID
-                )
-            )
+            // Always show the delivery status of the last sent message
+            val thisUsersSessionId = TextSecurePreferences.getLocalNumber(context)
+            val lastSentMessageId = mmsSmsDb.getLastSentMessageFromSender(message.threadId, thisUsersSessionId)
```
[the remark this answers is not in the document]
Comment (alansley) [reply app/src/main/java/org/thoughtcrime/securesms/conversation/v2/messages/VisibleMessageView.kt]: Done!

File: app/src/main/java/org/thoughtcrime/securesms/conversation/v2/ConversationAdapter.kt [hunk 2]
```diff
@@ -69,6 +70,12 @@ class ConversationAdapter(
         }
     }
 
+    // Static last sent message ID that's only updated when a message is sent or received (in
+    // `changeCursor`).
+    companion object {
+        var lastSentMessageId = -1L
```
[the remark this answers is not in the document]
Comment (alansley) [reply app/src/main/java/org/thoughtcrime/securesms/conversation/v2/ConversationAdapter.kt]: Adjusted to remove static property & send last message ID to `VisibleMessageView.bind`.

[the remark this answers is not in the document]
Comment (alansley) [reply app/src/main/java/org/thoughtcrime/securesms/conversation/v2/ConversationAdapter.kt]: Apologies, I wasn't quite sure what you were after, but this time I think we should be okay.

File: .gitmodules [hunk 3]
```diff
@@ -1,3 +1,4 @@
 [submodule "libsession-util/libsession-util"]
 	path = libsession-util/libsession-util
 	url = https://github.com/oxen-io/libsession-util.git
+	ignore = dirty
```
[the remark this answers is not in the document]
Comment (AL-Session) [reply .gitmodules:4]: Nah, just removed it and it all seems to still be playing nicely - so will push that removal & merge.

**Pull request merged**